src/applications/disability-benefits/all-claims/tests/components/ComboBox.unit.spec.jsx

```diff
@@ -23,22 +32,115 @@ describe('ComboBox', () => {
+
+      inputVaTextInput(input, searchTerm);
+
+      const filteredItems = fullStringSimilaritySearch(searchTerm, items);
+      const freeTextAndFilteredItemsLength = filteredItems.length + 1;
+
+      expect(listbox).to.have.length(freeTextAndFilteredItemsLength);
+    });
+
+    it('should render listbox items in alignment with string similarity search', () => {
+      const searchTerm = 'PT';
+      const searchResults = fullStringSimilaritySearch(searchTerm, items);
+
+      const { getAllByRole, getByTestId } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      inputVaTextInput(input, searchTerm);
+
+      const listboxItems = getAllByRole('option');
+
+      listboxItems.forEach((item, index) => {
+        if (index === 0) {
+          expect(item).to.contain.text(
+            `Enter your condition as "${searchTerm}"`,
+          );
+        } else {
+          expect(item).to.contain.text(searchResults[index - 1]);
+        }
+      });
+    });
   });
 
-  // can't find the input element inside web-component shadowDOM
-  // in order to test any other functionality here...
-  // it('should handle keyboard input', async () => {
-  //   const tree = render(<ComboBox {...defaultProps} />);
-  //   expect(tree.getByRole('listbox')).to.have.length(0);
-  //   const input = tree.getByLabelText('Test');
-  //   await userEvent.type(input, 'acl', { skipClick: true });
-  //   expect(tree.getByRole('listbox')).to.have.length(21);
-  // });
+  describe('User interactions', () => {
+    it('should highlight listbox item on mouse enter', () => {
+      const searchTerm = 'PT';
+      const { getAllByRole, getByTestId } = render(<ComboBox {...props} />);
+      const input = getByTestId('combobox-input');
+      inputVaTextInput(input, searchTerm);
+
+      const listboxItems = getAllByRole('option');
+      fireEvent.mouseEnter(listboxItems[0]);
+
+      expect(listboxItems[0]).to.have.class('cc-combobox__option--active');
+    });
+
+    it('should select listbox free text item on click and then listbox empty', () => {
+      const searchTerm = 'PT';
+      const { getAllByRole, getByRole, getByTestId } = render(
+        <ComboBox {...props} />,
+      );
+
+      const input = getByTestId('combobox-input');
+      inputVaTextInput(input, searchTerm);
+
+      const listboxItems = getAllByRole('option');
+      fireEvent.click(listboxItems[0]);
+
+      expect(input.value).to.eq(searchTerm);
+
+      const listbox = getByRole('listbox');
+      expect(listbox).to.have.length(0);
+    });
+
+    it('should select listbox item on click and then listbox empty', () => {
+      const searchTerm = 'Hear';
+      const { getByRole, getAllByRole, getByTestId } = render(
+        <ComboBox {...props} />,
+      );
+
+      const input = getByTestId('combobox-input');
+      inputVaTextInput(input, searchTerm);
+
+      const listboxItems = getAllByRole('option');
+      fireEvent.click(listboxItems[1]);
```

Comment:
Just a note that for the expanded component tests we'll want to consider keydown events as well